src/modules/flight_mode_manager/tasks/Transition/FlightTaskTransition.cpp

```diff
@@ -109,8 +118,10 @@ float FlightTaskTransition::computeBackTranstionPitchSetpoint()
 
 	float deceleration_setpoint = _param_vt_b_dec_mss;
 
-	if (_sub_position_sp_triplet.get().current.valid && _sub_vehicle_local_position.get().xy_global
-	    && position_xy.isAllFinite() && velocity_xy.isAllFinite()) {
+	const float max_hor_pos_uncertainty_limit = 10.f;
+
+	if (_eph < max_hor_pos_uncertainty_limit && _sub_position_sp_triplet.get().current.valid
```

Comment:
```suggestion
	if (_sub_vehicle_local_position.get().eph < max_hor_pos_uncertainty_limit && _sub_position_sp_triplet.get().current.valid
```
Then you can drop the overhead created with _eph. 
Possibly also a check for `_sub_vehicle_local_position.get().xy`_valid is needed.